cmd/collector/app/handler/grpc_handler_test.go

```diff
@@ -152,3 +168,167 @@ func TestPostSpansWithError(t *testing.T) {
+	return metadata.NewOutgoingContext(ctx, md)
+}
+
+func TestPostTenantedSpans(t *testing.T) {
+	tenantHeader := "x-tenant"
+	dummyTenant := "grpc-test-tenant"
+
+	processor := &mockSpanProcessor{}
+	server, addr := initializeGRPCTestServer(t, func(s *grpc.Server) {
+		handler := NewGRPCHandler(zap.NewNop(), processor,
+			tenancy.NewTenancyConfig(&tenancy.Options{
+				Enabled: true,
+				Header:  tenantHeader,
+				Tenants: []string{dummyTenant},
+			}))
+		api_v2.RegisterCollectorServiceServer(s, handler)
+	})
+	defer server.Stop()
+	client, conn := newClient(t, addr)
+	defer conn.Close()
+
+	ctxWithTenant := withMetadata(context.Background(), tenantHeader, dummyTenant, t)
+	ctxNoTenant := context.Background()
+	mdTwoTenants := metadata.Pairs()
+	mdTwoTenants.Set(tenantHeader, "a", "b")
+	ctxTwoTenants := metadata.NewOutgoingContext(context.Background(), mdTwoTenants)
+	ctxBadTenant := withMetadata(context.Background(), tenantHeader, "invalid-tenant", t)
+
+	withMetadata(context.Background(),
+		tenantHeader, dummyTenant, t)
+
+	tests := []struct {
+		name            string
+		ctx             context.Context
+		batch           model.Batch
+		mustFail        bool
+		expected        []*model.Span
+		expectedTenants map[string]bool
+	}{
+		{
+			name:  "valid tenant",
+			ctx:   ctxWithTenant,
+			batch: model.Batch{Process: &model.Process{ServiceName: "batch-process"}, Spans: []*model.Span{{OperationName: "test-op", Process: &model.Process{ServiceName: "bar"}}}},
+
+			mustFail:        false,
+			expected:        []*model.Span{{OperationName: "test-op", Process: &model.Process{ServiceName: "bar"}}},
+			expectedTenants: map[string]bool{dummyTenant: true},
+		},
+		{
+			name:  "no tenant",
+			ctx:   ctxNoTenant,
+			batch: model.Batch{Process: &model.Process{ServiceName: "batch-process"}, Spans: []*model.Span{{OperationName: "test-op", Process: &model.Process{ServiceName: "bar"}}}},
+
+			// Because NewGRPCHandler expects a tenant header, it will reject spans without one
+			mustFail:        true,
+			expected:        nil,
+			expectedTenants: nil,
+		},
+		{
+			name:  "two tenants",
+			ctx:   ctxTwoTenants,
+			batch: model.Batch{Process: &model.Process{ServiceName: "batch-process"}, Spans: []*model.Span{{OperationName: "test-op", Process: &model.Process{ServiceName: "bar"}}}},
+
+			// NewGRPCHandler rejects spans with multiple values for tenant header
+			mustFail:        true,
+			expected:        nil,
+			expectedTenants: nil,
+		},
+		{
+			name:  "invalid tenant",
+			ctx:   ctxBadTenant,
+			batch: model.Batch{Process: &model.Process{ServiceName: "batch-process"}, Spans: []*model.Span{{OperationName: "test-op", Process: &model.Process{ServiceName: "bar"}}}},
+
+			// NewGRPCHandler rejects spans with multiple values for tenant header
+			mustFail:        true,
+			expected:        nil,
+			expectedTenants: nil,
+		},
+	}
+	for _, test := range tests {
+		_, err := client.PostSpans(test.ctx, &api_v2.PostSpansRequest{
```

Comment:
the individual test cases should be executed separately via `t.Run`

Reply:
I followed the format of `TestPostSpans()`.  No tests in _cmd/collector/app/handler_ use `t.Run()`.